src/components/Loader/__tests__/loader-snapshots.jest.js

```diff
@@ -1,13 +1,94 @@
 import React from "react";
 import renderer from "react-test-renderer";
 import Loader from "../Loader";
+import styles from "./loader-snapshots.jest.module.scss";
 
-it("renders correctly with empty props", () => {
-  const tree = renderer.create(<Loader />).toJSON();
-  expect(tree).toMatchSnapshot();
-});
+describe("Loader renders correctly", () => {
+  it("with custom class name", () => {
+    const tree = renderer.create(<Loader size={Loader.sizes.MEDIUM} className="dummy-class-name" />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe("with size variants", () => {
+    it("with XS size", () => {
+      const tree = renderer.create(<Loader size={Loader.sizes.XS} />).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it("with SMALL size", () => {
+      const tree = renderer.create(<Loader size={Loader.sizes.SMALL} />).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it("with MEDIUM size", () => {
+      const tree = renderer.create(<Loader size={Loader.sizes.MEDIUM} />).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it("with LARGE size", () => {
+      const tree = renderer.create(<Loader size={Loader.sizes.LARGE} />).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it("with custom size from class", () => {
```

Comment:
this test and "with custom size from parent test" are not needed. because the snapshot not actually check the appearance on the loader, it just render the elements node tree according to the react components output. so, if the loader will not change it's size according to the class, the test still will pass.